### ISSUE TITLE
Chance to test solutions

### ISSUE THREAD

**Issue by acip** [hobbit-project/DEBS-GC-2018#45]
Hi @smirnp,

For the last couple of days the platform was very busy with very long running systems (not related to DEBS). We have systems running for hours. We understand that the platform is a shared resource for various benchmarks, but the current situation makes it close to impossible to be able to proper test our systems before the deadline.

Is it possible to "reserve" the platform for DEBS GC until the deadline? Or other solution you may think of as an "insider"?

Thank you!
Ciprian

**Comment by smirnp** [comment]
Hi @acip , unfortunately there is no real mechanism to prioritize executions except negotiations with other teams using the platform. 

If you looked into the SDK-based sample system - it allows you to execute the benchmark containers locally and also your system containers, so it largely helps during debugging. The only difference from online platform - is the resource pool (single machine vs 6 machines cluster).

**Comment by acip** [comment]
I know, but would the local benchmark have the latest ;labeled test tuples?

**Comment by eonica** [comment]
Dear @smirnp , @MichaelRoeder ,

Pavel, thank you for your answer, but in this phase is obviously more significant to do some final tests on the platform, for last minute tuning. There is also the system working time involved, which we cannot accurately reproduce locally. 

We have noticed that there are other benchmarks which are allocated more than 10000 seconds as initial running time, which would mean keeping the platform blocked for almost 3 hours. Although some experiments using these benchmarks were ended by the user, there are also recent experiments that took too much time. 

I assume there might be valuable debugging info collected in such long runs terminated with failure. However, would it be possible, considering the DEBS deadline and the need to use the platform, to temporarily limit a bit the maximum amount of time that a benchmark can take? (for me 10000 seconds seems a very long time). 

Thank you,
Emanuel

**Comment by smirnp** [comment]
For now I made online benchmark images publicly available, so you can replay it locally. 
So you may clone this repo and run the SampleSystemTest, benchmark images will be pulled automatically.
Benchmark images contain 20% of test data (used in the online benchmark) from the first data set labelled by improved (v8) algorithm, which will be used in the Final challenge.

**Comment by MichaelRoeder** [comment]
Hi all,

unfortunately there are several deadlines during these days (e.g., three other challenges running in parallel to DEBS GC). We are aware of the problem that all participants would like to test their systems as often as possible and we are trying to make sure that experiments that are not really necessary are postponed (if we know the person that would like to execute these experiments).

I agree that some situations could be improved. For example, I saw situations where a single system was listed around 10 times in the experiment queue with the same benchmark. Something like that could be prohibited in the future by limiting the number of experiments a user can have in the queue. However, the general problem (several people would like to use the platform at the same time) remains.

I understand your concerns regarding experiments which take several hours and I agree that we could have had a better management there (i.e., ask the person to start it in the evening instead of the middle of the day). I apologize for this mistake.

On the other hand side, I saw that the platform was not fully used yesterday and typically in the night, the platform is free as well. That does not mean that I am asking you to work in the night :wink: 
From my point of view a separation of your test runs (a local test during your development and a platform test once per day or every second day) could be a good strategy. So you can improve your system over the day and before you leave, you simply configure an experiment in the online platform which should be done until the next day.

However, we are very interested in supporting dedicated participants to make sure that the challenges are a success for everybody. So If you have ideas how we could further support the process of developing a solution for a given task of a challenge, don't hesitate to give (constructive) feedback. I could for example imagine that some people would like to get a notification (like an e-mail or something similar) when their experiment is done :thinking:

**Comment by nduy** [comment]
I agree with @MichaelRoeder that all of us should use the platform thoughtfully, with the awareness that there are other peoples also needs to perform their test. For the system which @MichaelRoeder noticed 10+ times listing, perhaps the owner should consider keeping only necessary runs and cancel the others. This will be highly appreciated as it leaves the chance for others to do their experiment. Thanks.
